### PR TITLE
Refactor of code base

### PR DESCRIPTION
**Pull request by linqcan**
Changes:
- No parsning in default.py only view related functionality
- Keep SVT related methods in a svt.py module
- Keep shared methods in a helper.py module
- Decrease the number of calls to fetchPage and parseDOM (improve speed)

### CODE REVIEW

**Comment by linqcan** [comment]
Possible methods that can be refactored as well:
- viewProgramsByLetter(letter)

Other things to consider:
- White spaces; keep it consistent.
- URL constants used for parsing; should be in SVT module?

File: default.py [hunk 1]
```diff
@@ -31,7 +31,6 @@
 MODE_VIEW_EPISODES = "view_episodes"
 MODE_VIEW_CLIPS = "view_clips"
 
-BASE_URL = "http://www.svtplay.se"
 SWF_URL = "http://www.svtplay.se/public/swf/video/svtplayer-2013.02.swf"
```
Comment (nilzen) [inline default.py]: Shouldn't this go into the svt module? Perhaps most of the constants should go there...?

File: default.py [hunk 2]
```diff
@@ -105,7 +101,7 @@ def viewStart():
 
 
 def viewChannels():
-  channels = svt.getChannels(BASE_URL + URL_CHANNELS)
+  channels = svt.getChannels(URL_CHANNELS)
```
Comment (nilzen) [inline default.py]: Imo it would be cleaner with svt.getChannels()

File: default.py (hunk 1 above)
Comment (linqcan) [inline default.py]: Yep, agree (see comment in PR).

File: default.py [hunk 3]
```diff
@@ -117,7 +113,7 @@ def viewChannels():
   
 
 def viewAtoO():
-  html = helper.getPage(BASE_URL + URL_A_TO_O)
+  html = svt.getPage(URL_A_TO_O)
```
Comment (nilzen) [inline default.py]: svt.getAtoO() etc for the rest of the functions and inside it calls svt.getPage(URL_A_TO_O)

File: default.py (hunk 2 above)
Comment (linqcan) [inline default.py]: Agree (see email from last week)

File: default.py (hunk 3 above)
Comment (linqcan) [inline default.py]: Agree, (see email from last week)

**Comment by nilzen** [comment]
Other than the view related changes discussed in email it looks nice!

**Comment by linqcan** [comment]
Thanks! Will do the changes within the next few days.

**Comment by linqcan** [comment]
Commit updated with discussed changes!

File: default.py [hunk 4]
```diff
@@ -31,28 +31,19 @@
...
-SWF_URL = "http://www.svtplay.se/public/swf/video/svtplayer-2013.02.swf" 
 JSON_SUFFIX = "?output=json"
 
-URL_A_TO_O = "/program"
-URL_CATEGORIES = "/kategorier"
-URL_CHANNELS = "/kanaler"
 URL_TO_LATEST = "?tab=episodes&sida=1"
 URL_TO_LATEST_NEWS = "?tab=news&sida=1"
 URL_TO_RECOMMENDED = "?tab=recommended&sida=1"
 URL_TO_SEARCH = "/sok?q="
```
Comment (nilzen) [inline default.py]: What about these?

Comment (linqcan) [inline default.py]: They are needed in this module, but could be moved to the SVT module and be accessed by svt.URL_TO_LATEST. What do you think?

Comment (nilzen) [inline default.py]: Since they are SVT site specific I would move them as well. Just to remove all SVT site dependencies in this file.

Comment (linqcan) [inline default.py]: Agreed!

**Comment by linqcan** [comment]
Update. Moved more constants.

**Comment by nilzen** [comment]
I like the new style, can't test the function now. But if you have tested it, go ahead and merge!

**Comment by linqcan** [comment]
Tested all combinations of settings I could think of on XBMC 12.0 (Linux). Tested OK.

**Comment by linqcan** [comment]
The last update before the merge involved changing ugly "...replace" with a regular expression substitution instead. Line 108 in helper.py.

**Comment by nilzen** [comment]
Well done! :)

**Comment by linqcan** [comment]
Tack!

**Pull request merged**